compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -166,6 +166,7 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     let llfn = cx.get_fn(instance);
 
     let mir = cx.tcx().instance_mir(instance.def);
+    drop(std::hint::black_box(mir.clone()));
```

Comment:
Are you certain that you're copying `mir::Body` and not `&mir::Body`?

Reply:
I am not certain.

Reply:
Actually, I'm pretty certain now. `&mir::Body` will match against `Clone::clone` without applying autoref first, which clones `mir::Body`.

Reply:
Normally I'd just assume you know what you're talking about, but this result is quite remarkable if true, so I'm going to be doubly sure.

Reply:
yep :+1: there's a chance there's something *else* weird happening here too.

Reply:
Calling `(&T).clone()` should call `clone()` on `T` (even though `&T` also implements `Clone`, which is true for all types).
I would be more worried about the black box. Maybe you could try cloning the `mir` variable into `mir2`, and then continue using `mir2` in the code? (or just shadow the original variable).

Reply:
`mir::Body::clone` should avoid the ambiguity.